app/src/main/java/swati4star/createpdf/fragment/FAQFragment.java

```diff
@@ -24,44 +25,83 @@
 
     private FAQAdapter mFaqAdapter;
     private List<FAQItem> mFaqs;
+    private List<FAQItem> mFaqsCopy;
     private Context mContext;
+    private SearchView mSearchView;
 
     @BindView(R.id.recycler_view_faq)
     RecyclerView mFAQRecyclerView;
 
+
     public FAQFragment() {
         // Required empty public constructor
     }
 
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
 
         View view = inflater.inflate(R.layout.fragment_faq, container, false);
 
+        mSearchView =  view.findViewById(R.id.searchView);
+        mSearchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                filterFaq(query);
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newQuery) {
+                filterFaq(newQuery);
+                return true;
+            }
+        });
+
         ButterKnife.bind(this, view);
         mContext = view.getContext();
 
         initFAQs();
-
         initFAQRecyclerView();
 
         return view;
     }
 
+    /**
+     * @param text - This is the search tet entered in the search box.
+     * Simply filtering out the questions that contains the given search query.
+     */
+    public void filterFaq(String text) {
+        mFaqs.clear();
+        if (text.isEmpty())
+            mFaqs.addAll(mFaqsCopy);
+        else {
+            System.out.println(text);
+            text = text.toLowerCase();
+
+            for (FAQItem faq : mFaqsCopy) {
+                if (faq.getQuestion().toLowerCase().contains(text)) {
+                    mFaqs.add(faq);
+                }
+            }
+        }
+        mFaqAdapter.notifyDataSetChanged();
+    }
+
     /**
      * Initializes the FAQs questions and answers by populating data from the
      * strings.xml file and adding it to the custom FAQItem Model.
      */
     private void initFAQs() {
         mFaqs = new ArrayList<>();
+        mFaqsCopy = new ArrayList<>();
         String[] questionAnswers = mContext.getResources().getStringArray(R.array.faq_question_answers);
         FAQItem faqItem;
         for (String questionAnswer : questionAnswers) {
             String[] questionAnswerSplit = questionAnswer.split("#####");
             faqItem = new FAQItem(questionAnswerSplit[0], questionAnswerSplit[1]);
             mFaqs.add(faqItem);
+            mFaqsCopy.add(faqItem);
```

Comment:
instead of adding items to the copy one by one.

why not initialize it at once after the loop:

`mFaqsCopy = mFaqs;`

Reply:
Hi @Swati4star 

This will create problems, when I will clear the mFaqs in filterFAQs method, it will even clear mFaqsCopy, which is not desirable.
So, I initialized and then added all the FAQItems.
Reason: 
 `mFaqsCopy = mFaqs,` will just create a reference and not a deep copy.

Correct me if I am wrong or if you were expecting a different implementation.